src/main/java/billy/command/Command.java

```diff
@@ -0,0 +1,34 @@
+package billy.command;
+
+import billy.task.TaskList;
+import billy.ui.Ui;
+
+
+/**
+ * Represents an abstract command that can be executed in the application.
+ *
+ * <p>Each {@code Command} encapsulates a user input string and defines behavior
+ * through its {@link #execute(TaskList, Ui)} method. Subclasses should implement
+ * this method to specify how the command manipulates the {@code TaskList},
+ * interacts with the {@code Ui}, and potentially signals program termination.</p>
+ */
+public abstract class Command {
+    protected Boolean isExit;
```

Comment:
```suggestion
    protected boolean isExit;
```
If you prefer, maybe you would want to standardise using primitive types over wrapper classes? Same with method shouldExit().

Reply:
Boolean allows for null checks, but it is a valid point